coretex/utils/docker.py

```diff
@@ -198,3 +199,57 @@ def getLogs(name: str, tail: Optional[int], follow: bool, timestamps: bool) -> N
         runCommand.append("-f")
 
     command(runCommand)
+
+
+def isDockerDesktop() -> bool:
+    try:
+        _, output, _ = command(["docker", "context", "show"], ignoreStdout = True, ignoreStderr = True)
+        return output.strip() == "desktop-linux"
+    except:
+        return False
+
+
+def isDaemonFileUpdated() -> bool:
+    daemonFile = "/etc/docker/daemon.json"
+    cGroupFix = "native.cgroupdriver=cgroupfs"
+
+    if not Path(daemonFile).exists():
+        return False
+
+    with open(daemonFile, "r") as file:
+        try:
+            config = json.load(file)
+            execOpts = config.get("exec-opts", [])
+            return cGroupFix in execOpts
+        except json.JSONDecodeError:
+            return False
+
+
+def updateDaemonFile() -> None:
+    daemonFile = "/etc/docker/daemon.json"
+    cGroupFix = "native.cgroupdriver=cgroupfs"
+    config: Dict[str, Any] = {}
+
+    if not Path(daemonFile).exists():
+        config = {}
+
+    with open(daemonFile, "r") as file:
+        try:
+            config = json.load(file)
+        except json.JSONDecodeError:
+            config = {}
+
+    execOpts: List[str] = config.get("exec-opts", [])
+    execOpts.append(cGroupFix)
+    config["exec-opts"] = execOpts
+
+    with tempfile.NamedTemporaryFile("w", delete = False) as tempFile:
```

Comment:
`delete` should stay `True`, right? You just place the `command` call into the `with` as well.